plugins/chat/assets/javascripts/discourse/components/chat/thread-list.js

```diff
@@ -53,6 +55,57 @@ export default class ChatThreadList extends Component {
     return !!this.args.channel;
   }
 
+  @action
+  subscribe() {
+    this.#unsubscribe();
+
+    this.messageBus.subscribe(
```

Comment:
Another case where the composition would be nicer...if we could do something like this:

```javascript
import { chatChannelMessageBus } from "lib/chat/message-bus";

@action
subscribe() {
  this.deleteMessageSubscription = channelMessageBus.subscribeToDelete(this.args.channel, this.handleDeleteMessage);
 this.restoreMessageSubscription = channelMessageBus.subscribeToRestore(this.args.channel, this.handleRestoreMessage);
}

#unsubscribe() {
   this.deleteMessageSubscription.unsubscribe();
   this.restoreMessageSubscription.unsubscribe();
}
```

This would benefit us in the base pane subscription situation we are in to, to remove inheritance.

Reply:
Yes that's an interesting idea

Reply:
Maybe we could do that refactor in another PR?

Reply:
Yes no hurries